### PR TITLE
Add customizable media link; handle protocol-less media

### PR DESCRIPTION
**Pull request by OpportunityLiu**
#337 
A new interface `IHostProvider` is added for media link customization.

### CODE REVIEW

**Comment by OpportunityLiu** [comment]
#135

File: src/Markdig/Extensions/MediaLinks/MediaLinkExtension.cs [hunk 1]
```diff
@@ -55,18 +54,28 @@ private bool TryLinkInlineRenderer(HtmlRenderer renderer, LinkInline linkInline)
             }
 
             Uri uri;
+            bool isSchemaRelative = false;
             // Only process absolute Uri
             if (!Uri.TryCreate(linkInline.Url, UriKind.RelativeOrAbsolute, out uri) || !uri.IsAbsoluteUri)
             {
-                return false;
+                // see https://tools.ietf.org/html/rfc3986#section-4.2
+                // since reletive uri doesn't support many properties, "http" is used as a placeholder here.
```
Comment (leotsarev) [inline src/Markdig/Extensions/MediaLinks/MediaLinkExtension.cs]: relative

File: src/Markdig/Extensions/MediaLinks/HostProviderBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        public static IHostProvider Create(string hostPrefix, Func<Uri, string> handler, bool allowFullScreen = true, string iframeClass = null)
+        {
+            if (string.IsNullOrEmpty(hostPrefix))
+                throw new ArgumentException("hostPrefix is null or empty.", nameof(hostPrefix));
+            if (handler == null)
+                throw new ArgumentNullException(nameof(handler));
+            return new DelegateProvider { HostPrefix = hostPrefix, Delegate = handler, AllowFullScreen = allowFullScreen, Class = iframeClass };
+        }
+
+        public static Dictionary<string, IHostProvider> KnownHosts { get; }
```
Comment (xoofx) [inline src/Markdig/Extensions/MediaLinks/HostProviderBuilder.cs]: Could you make this property internal? 

```suggestion
        internal static Dictionary<string, IHostProvider> KnownHosts { get; }
```

**Comment by xoofx** [comment]
Thanks!

**Pull request merged**